crates/ruff_dev/src/generate_rules_table.rs

```diff
@@ -22,7 +22,7 @@ fn generate_table(table_out: &mut String, rules: impl IntoIterator<Item = Rule>,
 
         #[allow(clippy::or_fun_call)]
         table_out.push_str(&format!(
-            "| {}{} | {} | {} | {} |",
+            "| <div id='{0}{1}'>{0}{1}</div> | {2} | {3} | {4} |",
```

Comment:
You can use `mkdocs` attribute-lists markdown extension to set the id without the need for a wrapper `div`
```suggestion
            "| {0}{1} {{ #{0}{1} }} | {2} | {3} | {4} |",
```